block.go

```diff
@@ -243,7 +243,11 @@ func (p *parser) prefixHeader(out *bytes.Buffer, data []byte) int {
 	}
 	if end > i {
 		if id == "" && p.flags&EXTENSION_AUTO_HEADER_IDS != 0 {
-			id = sanitized_anchor_name.Create(string(data[i:end]))
+			if p.sanitizedAnchorNameOverride == nil {
+				id = sanitized_anchor_name.Create(string(data[i:end]))
+			} else {
+				id = p.sanitizedAnchorNameOverride(string(data[i:end]))
+			}
```

Comment:
I think it'd be cleaner to ensure `p.sanitizedAnchorNameOverride` is never nil (by giving it a default value when there's no override), so the code here can be just:

```Go
id = p.sanitizedAnchorNameOverride(string(data[i:end]))
```

At that point, you'd want to rename `sanitizedAnchorNameOverride` to just `sanitizedAnchorName`.

If an override is provided, its value will be set to:

```Go
sanitizedAnchorName = sanitized_anchor_name.Create
```